### PR TITLE
Fix links to examples in main README and landing page

### PR DESCRIPTION
**Pull request by mziccard**

### CODE REVIEW

**Comment by lesv** [comment]
LGTM - though you might wish to add https://github.com/GoogleCloudPlatform/java-docs-samples/tree/master/appengine/guestbook-cloud-datastore to the list.

(Didn't realize you were tracking these things, or I'd have fixed the links for you.

File: README.md [hunk 1]
```diff
@@ -65,16 +65,17 @@ Example Applications
...
+- [`Flexible Environment/Datastore example`](https://github.com/GoogleCloudPlatform/java-docs-samples/tree/master/flexible/datastore) - A simple app that uses Cloud Datastore to list the last 10 IP addresses that visited your site.
   - Read about how to run the application [here](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/managed_vms/README.md).
-- [`Flexible Environment/Storage example`](https://github.com/GoogleCloudPlatform/java-docs-samples/tree/master/managed_vms/cloudstorage) - An app that uploads files to a public Cloud Storage bucket on the App Engine Flexible Environment runtime.
+- [`Flexible Environment/Storage example`](https://github.com/GoogleCloudPlatform/java-docs-samples/tree/master/flexible/cloudstorage) - An app that uploads files to a public Cloud Storage bucket on the App Engine Flexible Environment runtime.
+- [`GuestBook`](https://github.com/GoogleCloudPlatform/java-docs-samples/tree/master/appengine/guestbook-cloud-datastore) - An App Engine guestbook that uses Cloud Datastore.
```
Comment (lesv) [inline README.md]: App Engine standard

Comment (lesv) [inline README.md]: There are two products App Engine Standard & Flexible

Comment (mziccard) [inline README.md]: Can't the example run also on AE flexible?

Comment (mziccard) [inline README.md]: Also in the example's README itself there's no mention of Standard or Flexible

Comment (lesv) [inline README.md]: We should probably talk about samples in categories:
1. Pure Java - works from the command line, should work on GCP, GKE
2. GAE Flex - Java 8, deployable, includes gRPC samples
3. GAE Standard - REST based APIs only

Comment (lesv) [inline README.md]: There are differences between GAE Standard and Flexible that prevents a sample from running on both platforms.

Comment (mziccard) [inline README.md]: Added 3a1fd9e to address this

**Pull request merged**